repo/db/peers.go

```diff
@@ -111,18 +111,40 @@ func (c *PeerDB) Get(id string) *pb.Peer {
 	return res[0]
 }
 
-func (c *PeerDB) GetBest(id string) *pb.Peer {
+func (c *PeerDB) GetBestUser(id string) *pb.User {
```

Comment:
This tries to get the best info for a peer by checking against other peers that have the same address. Related to the post-sync name and avatar wipe of your _own_ contact, this should fix that same issue cropping within the feed.